### PR TITLE
Restructure project directories

### PR DESCRIPTION
**Pull request by tzheleznyak**
### What does this PR do?
As part of the refactoring process this PR organizes the directories to a stracture based on the porpuse of the files and not on the conjur command they preform. This will make the codebase more easier to maintain. You can read more here:
https://github.com/cyberark/conjur-api-python3/blob/design/general-refactorings/design/general_refactorings.md#organize-directory-structure

### What ticket does this PR close?
Is part of the general refactor from here https://github.com/cyberark/conjur-api-python3/issues/121

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: conjur/conjur_api/conjur_client.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ class ConfigException(Exception):
     use these parameters defined in this class to initialize our Python
     SDK in their code.
     """
-class Client():
+class ConjurClient():
```
Comment (liavyona) [inline conjur/conjur_api/conjur_client.py:40]: It is kind of redandunt to change the name to Conjur* because the library name is conjur. I can use like `conjur.api.Api` which is very clear to me that this is the conjur API (so as client, config etc.)

Comment (tzheleznyak) [reply conjur/conjur_api/conjur_client.py]: Done

Comment (liubovra) [reply conjur/conjur_api/conjur_client.py]: I agree with Tamir, because in code we don't see the full package and only the Class name, so now in our testing code I only see Client and its not really readable to understand which client

**Comment by Tovli** [comment]
@tzheleznyak - great initiative
Can you please add details to the PR? what issue are you trying to resolve? is there a naming standard you are implementing? what is the affect on UX?
Both the issue and the PR lack this info

**Comment by tzheleznyak** [comment]
> @tzheleznyak - great initiative
> Can you please add details to the PR? what issue are you trying to resolve? is there a naming standard you are implementing? what is the affect on UX?
> Both the issue and the PR lack this info

The link is attached there but i added small description that this is part of the refactoring process

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Update version from 0.x.x to 7.0.0 to prep for the GA release
 - Update help screens according to [these guidelines](https://ljfz3b.axshare.com/#id=yokln4&p=conjur_main_help&g=1).
   [cyberark/conjur-api-python3#92](https://github.com/cyberark/conjur-api-python3/issues/92)
+- Directory Structure Changed [dir changes](https://github.com/cyberark/conjur-api-python3/blob/design/general-refactorings/design/general_refactorings.md#organize-directory-structure)
```
Comment (sigalsax) [inline CHANGELOG.md:30]: - Directory structure has been changed. See [design document](https://github.com/cyberark/conjur-api-python3/blob/design/general-refactorings/design/general_refactorings.md#organize-directory-structure) for more details. [cyberark/conjur-api-python3#xyz](https://github.com/cyberark/conjur-api-python3/issues/xyz)

A couple of things. 
1. Please add the GH issue
2. I would also maybe mention a warning to the user about his this can impact them

File: conjur/api/client.py [hunk 3]
```diff
@@ -14,15 +14,15 @@
...
-from conjur.api import Api
+from conjur.api.api import Api
 from conjur.config import Config as ApiConfig
 from conjur.constants import DEFAULT_NETRC_FILE
-from conjur.init.init_controller import InitController
-from conjur.init.init_logic import InitLogic
-from conjur.init.conjurrc_data import ConjurrcData
-from conjur.credentials_from_file import CredentialsFromFile
+from conjur.controllers.init_controller import InitController
+from conjur.logics.init_logic import InitLogic
```
Comment (sigalsax) [inline conjur/api/client.py:21]: `logics` isn't a word.. seems a bit weird. So I would maybe write logic

Comment (sigalsax) [reply conjur/api/client.py:21]: maybe leave everything singular. for example, api isn't apis

File: conjur/api/client.py [hunk 4]
```diff
@@ -105,19 +105,19 @@ def __init__(self,
...
+                                  http_debug=http_debug,
+                                  login_id=login_id,
+                                  ssl_verify=ssl_verify,
+                                  **loaded_config)
         elif password:
             logging.debug("Creating API key with login ID/password combo...")
             self._api = Api(http_debug=http_debug,
-                            ssl_verify=ssl_verify,
-                            **loaded_config)
+                                  ssl_verify=ssl_verify,
```
Comment (sigalsax) [inline conjur/api/client.py:115]: why this indentation? Should be inline with the above params

File: conjur/cli.py [hunk 5]
```diff
@@ -561,7 +570,7 @@ def handle_logout_logic(cls, ssl_verify=True):
         """
         Method that wraps the logout call logic
         """
-        credentials = CredentialsFromFile(DEFAULT_NETRC_FILE)
+        credentials = NetrcWrapper(DEFAULT_NETRC_FILE)
```
Comment (sigalsax) [inline conjur/cli.py:573]: Purposefully didn't want to name is Netrc-something. B/c today it is netrc, sometomorrow it something else

Comment (sigalsax) [reply conjur/cli.py:573]: I am also not sure why it is a wrapper? What is your thoughts here

File: conjur/controllers/logout_controller.py [hunk 6]
```diff
@@ -14,7 +14,7 @@
 
 # Internals
 from conjur.constants import DEFAULT_NETRC_FILE, DEFAULT_CONFIG_FILE
-from conjur.init import ConjurrcData
+from conjur.data_objects.conjurrc_data import ConjurrcData
```
Comment (sigalsax) [inline conjur/controllers/logout_controller.py:17]: maybe just data_object?

File: test/test_config/missing_machine_netrc [hunk 7]
```diff
@@ -4,4 +4,4 @@ machine https://randommachine/
 
 machine https://anotherrandommachine
   login someadmin
-  password conjurapikey
+  password Apikey
```
Comment (sigalsax) [inline test/test_config/missing_machine_netrc:7]: why?

File: conjur/controllers/logout_controller.py (hunk 6 above)
Comment (tzheleznyak) [reply conjur/controllers/logout_controller.py:17]: folder names should be in plural so its data_objects

File: test/test_config/missing_machine_netrc (hunk 7 above)
Comment (tzheleznyak) [reply test/test_config/missing_machine_netrc:7]: it was chanhged by mistake with ctrl shift r . Good Catch. I take it back

File: conjur/cli.py (hunk 5 above)
Comment (tzheleznyak) [reply conjur/cli.py:573]: But it has function build_netrc. not build_configfile. so we need to change the function name also ?

File: conjur/api/client.py (hunk 3 above)
Comment (tzheleznyak) [reply conjur/api/client.py:21]: Will change to singular

File: CHANGELOG.md (hunk 2 above)
Comment (tzheleznyak) [reply CHANGELOG.md:30]: Done

File: conjur/api/client.py (hunk 3 above)
Comment (tzheleznyak) [reply conjur/api/client.py:21]: changed to singular and logic

File: conjur/api/client.py (hunk 4 above)
Comment (tzheleznyak) [reply conjur/api/client.py:115]: Fixed

File: CHANGELOG.md [hunk 8]
```diff
@@ -27,7 +27,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Update version from 0.x.x to 7.0.0 to prep for the GA release
 - Update help screens according to [these guidelines](https://ljfz3b.axshare.com/#id=yokln4&p=conjur_main_help&g=1).
   [cyberark/conjur-api-python3#92](https://github.com/cyberark/conjur-api-python3/issues/92)
-- Directory Structure Changed [dir changes](https://github.com/cyberark/conjur-api-python3/blob/design/general-refactorings/design/general_refactorings.md#organize-directory-structure)
+- Directory structure has been changed. See [design document](https://github.com/cyberark/conjur-api-python3/blob/design/general-refactorings/design/general_refactorings.md#organize-directory-structure) for more details.
```
Comment (sigalsax) [inline CHANGELOG.md:30]:
```
Users who import the SDK should change their import statement from `conjur.client import Client` to  `from conjur.api.client import Client`[cyberark/conjur-api-python3#121](https://github.com/cyberark/conjur-api-python3/issues/121)
```

Comment (sigalsax) [reply CHANGELOG.md:30]: @orenbm  👀 
We are changing project directory which means this is a breaking change for those that import the Client in their projects. How is the recommended changelog entry?

File: conjur/api/client.py [hunk 9]
```diff
@@ -17,13 +17,14 @@
 from conjur.api.api import Api
 from conjur.config import Config as ApiConfig
 from conjur.constants import DEFAULT_NETRC_FILE
-from conjur.controllers.init_controller import InitController
-from conjur.logics.init_logic import InitLogic
-from conjur.data_objects.conjurrc_data import ConjurrcData
+from conjur.controller.init_controller import InitController
+from conjur.logic.init_logic import InitLogic
+from conjur.data_object.conjurrc_data import ConjurrcData
 from conjur.wrappers.netrc_wrapper import NetrcWrapper
```
Comment (sigalsax) [inline conjur/api/client.py:23]: why netrc_wrapper? 
I think we should rethink this name two fold 1. should we call it netrc? something so so specific 2. should it be a wrapper?

Please explain what you think here (:

File: conjur/api/client.py [hunk 10]
```diff
@@ -37,6 +38,8 @@ class ConfigException(Exception):
     use these parameters defined in this class to initialize our Python
     SDK in their code.
     """
+
+
```
Comment (sigalsax) [inline conjur/api/client.py:42]: extra two lines

File: CHANGELOG.md (hunk 8 above)
Comment (orenbm) [reply CHANGELOG.md:30]: well, users would know that the API has been broken by the version as its MAJOR should change.
However, iirc this is version 0 so we don't need to do that if we don't want to. I think that on version 0 users should assume that the API may break.

Maybe we can write that this change breaks the API?

Comment (orenbm) [reply CHANGELOG.md:30]: we did something similar here: https://github.com/cyberark/conjur-authn-k8s-client/blob/master/CHANGELOG.md#changed

File: conjur/api/client.py (hunk 10 above)
Comment (tzheleznyak) [reply conjur/api/client.py:42]: Remvoed

File: conjur/api/client.py (hunk 9 above)
Comment (tzheleznyak) [reply conjur/api/client.py:23]: Renamed to CredentialsFromFile  and moved to logic folder

File: conjur/logic/credentials_from_file.py [hunk 11]
```diff
@@ -16,12 +16,15 @@
 # Internals
 from conjur.constants import DEFAULT_NETRC_FILE
 
+
+# TODO: Break this class to reader and writer classes and divide each one to generic class for read/write and  class for netrc that inherits it.
```
Comment (BradleyBoutcher) [inline conjur/logic/credentials_from_file.py:20]: Would it be better to file an issue that references this line instead of adding a `TODO`?

File: CHANGELOG.md (hunk 8 above)
Comment (BradleyBoutcher) [reply CHANGELOG.md:30]: Unless we post it somewhere, I don't think users would assume breaking changes could regularly occur without a major number change. If this change is breaking, it should absolutely get a major version bump on the next release.

**Review by BradleyBoutcher — COMMENTED** [review]
Looking good - just adding some things to consider

File: CHANGELOG.md [hunk 12]
```diff
@@ -27,6 +27,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Update version from 0.x.x to 7.0.0 to prep for the GA release
 - Update help screens according to [these guidelines](https://ljfz3b.axshare.com/#id=yokln4&p=conjur_main_help&g=1).
   [cyberark/conjur-api-python3#92](https://github.com/cyberark/conjur-api-python3/issues/92)
+- Directory structure has been changed. See [design document](https://github.com/cyberark/conjur-api-python3/blob/design/general-refactorings/design/general_refactorings.md#organize-directory-structure) for more details.
+  Users that import `Api` and `Client` should change their import statement to  api dir now to the import.
```
Comment (sigalsax) [inline CHANGELOG.md:31]: @tzheleznyak Let's add 'This is a breaking change. Users that import...'

File: conjur/logic/credentials_from_file.py (hunk 11 above)
Comment (tzheleznyak) [reply conjur/logic/credentials_from_file.py:20]: Added this issue and removed TODO
https://github.com/cyberark/conjur-api-python3/issues/173

File: CHANGELOG.md (hunk 12 above)
Comment (tzheleznyak) [reply CHANGELOG.md:31]: Fixed

**Comment by tzheleznyak** [comment]
@BradleyBoutcher Soon there will be Python CLI realse anyway with a version bump so it will be part of it

File: CHANGELOG.md (hunk 8 above)
Comment (tzheleznyak) [reply CHANGELOG.md:30]: Fixed

Comment (tzheleznyak) [reply CHANGELOG.md:30]: It will be a part of the coming soon CLI release so no need for special version bump for it.

File: CHANGELOG.md [hunk 13]
```diff
@@ -27,6 +27,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Update version from 0.x.x to 7.0.0 to prep for the GA release
 - Update help screens according to [these guidelines](https://ljfz3b.axshare.com/#id=yokln4&p=conjur_main_help&g=1).
   [cyberark/conjur-api-python3#92](https://github.com/cyberark/conjur-api-python3/issues/92)
+- Directory structure has been changed. See [design document](https://github.com/cyberark/conjur-api-python3/blob/design/general-refactorings/design/general_refactorings.md#organize-directory-structure) for more details.
```
Comment (InbalZilberman) [inline CHANGELOG.md:30]: Nice to have - how about using the word refactored instead of changed?

File: conjur/api/client.py [hunk 14]
```diff
@@ -14,15 +14,16 @@
 import netrc
 
 from Utils.utils import Utils
-from conjur.api import Api
+from conjur.api.api import Api
```
Comment (InbalZilberman) [inline conjur/api/client.py:17]: hmmm is there any way to change one of the api.api?
looks a bit strange...

File: conjur/cli.py (hunk 5 above)
Comment (Tovli) [reply conjur/cli.py]: I agree with Sigal, and with your addition :)
We should abstract any external resource and use business phrases to describe the action

File: CHANGELOG.md (hunk 13 above)
Comment (tzheleznyak) [reply CHANGELOG.md:30]: Changed to refactored

File: conjur/api/client.py (hunk 14 above)
Comment (tzheleznyak) [reply conjur/api/client.py:17]: I moved it to init file so in here it will be 
`from conjur.api import api`

File: CHANGELOG.md (hunk 12 above)
Comment (sigalsax) [reply CHANGELOG.md:31]: Please delete this line " Users that import `Api` and `Client` should change their import statement to  api dir now to the import." as this is already stated below

Should be 
Directory structure has been refactored. See [design document](https://github.com/cyberark/conjur-api-python3/blob/master/design/general_refactorings.md) for more details. This is a breaking change. Users who import the SDK in their projects should change their import statement from 
  `conjur.client import Client` will change it to `from conjur.api.client import Client`
  [cyberark/conjur-api-python3#121](https://github.com/cyberark/conjur-api-python3/issues/121)

File: conjur/api/__init__.py [hunk 15]
```diff
@@ -0,0 +1 @@
+from conjur.api.api import Api
```
Comment (sigalsax) [inline conjur/api/__init__.py:1]: new line

File: conjur/api/api.py [hunk 16]
```diff
@@ -13,8 +13,8 @@
 from datetime import datetime, timedelta
 
 # Internals
-from conjur.endpoints import ConjurEndpoint
-from conjur.http_wrapper import HttpVerb, invoke_endpoint
+from conjur.api.endpoints import ConjurEndpoint
```
Comment (sigalsax) [inline conjur/api/api.py:16]: GENERAL NOTE, is there a way that you can manually test the initialization of the Client. Ensure you can still perform basic actions (like list, set variable)?

File: conjur/api/api.py [hunk 17]
```diff
@@ -13,8 +13,8 @@
...
+from conjur.wrappers.http_wrapper import HttpVerb, invoke_endpoint
```
Comment (sigalsax) [inline conjur/api/api.py:17]: wrappers should be wrapper to conform with rest of directory structure

File: conjur/api/ssl_client.py [hunk 18]
```diff
@@ -20,7 +20,7 @@
 }
 
 # pylint: disable=too-few-public-methods
-class SSLService:
+class SSLClient:
     """
     SSLService
```
Comment (sigalsax) [inline conjur/api/ssl_client.py:25]: GENERAL NOTE
Please change all docstrings to their appropriate, updated names. For example, SSLService here should now be SSLClient

File: conjur/cli.py [hunk 19]
```diff
@@ -19,21 +19,30 @@
...
+from conjur.data_object.variable_data import VariableData
+from conjur.logic.list_logic import ListLogic
+from conjur.logic.login_logic import LoginLogic
+from conjur.logic.logout_logic import LogoutLogic
+from conjur.logic.policy_logic import PolicyLogic
+from conjur.logic.user_logic import UserLogic
+from conjur.logic.variable_logic import VariableLogic
 from conjur.version import __version__
 
 # pylint: disable=too-many-statements
```
Comment (sigalsax) [inline conjur/cli.py:48]: all new file directory __init__ should not be empty IMO. They should contain the controllers that they represent from example `import init_controller from InitController`. 

Please also add a docstring explaining what each directory contains

"""
Controller module

This metafile includes all controllers required in this project
"""

same for data_object and same for api, etc etc

File: test/test_unit_init_controller.py [hunk 20]
```diff
@@ -116,7 +116,7 @@ def test_user_confirms_force_overwrites_writes_cert_to_file(self, mock_input, mo
     '''
     # The conjurrc file exists and the CLI prompts
     # if the user wants to overwrite
-    @patch('conjur.init.init_logic')
+    @patch('conjur.logic.init_logic')
```
Comment (sigalsax) [inline test/test_unit_init_controller.py:119]: is this the only place this change in directory applies?

**Review by sigalsax — CHANGES_REQUESTED** [review]
Please see my comments on the PR. They need to be addressed before we can merge. Overall good work!

File: conjur/api/api.py (hunk 17 above)
Comment (tzheleznyak) [reply conjur/api/api.py:17]: Changed to wrapper

File: conjur/api/ssl_client.py (hunk 18 above)
Comment (tzheleznyak) [reply conjur/api/ssl_client.py:25]: Changed docstring to SSLClient.
These is the only class its name is changed

File: test/test_unit_init_controller.py (hunk 20 above)
Comment (tzheleznyak) [reply test/test_unit_init_controller.py:115]: There are many patches that changed becuase of the dir change

File: conjur/cli.py (hunk 19 above)
Comment (tzheleznyak) [reply conjur/cli.py:48]: Added

File: conjur/api/api.py (hunk 17 above)
Comment (tzheleznyak) [reply conjur/api/api.py]: Done

File: conjur/api/__init__.py (hunk 15 above)
Comment (tzheleznyak) [reply conjur/api/__init__.py]: Added lines between comment and imports in the init files

File: conjur/logic/__init__.py [hunk 21]
```diff
@@ -0,0 +1,14 @@
+"""
+logic package
+
+This package contain the logic classes
+"""
+
+from conjur.logic.credentials_from_file import CredentialsFromFile
```
Comment (sigalsax) [inline conjur/logic/__init__.py:7]: why is this part of logic?

File: conjur/wrapper/__init__.py [hunk 22]
```diff
@@ -0,0 +1,5 @@
+"""
+wrapper package
```
Comment (sigalsax) [inline conjur/wrapper/__init__.py:2]: Wrapper

don't need 'package also'

File: conjur/wrapper/__init__.py [hunk 23]
```diff
@@ -0,0 +1,5 @@
...
+
+This package contain classes that are wrapping third party libraries
```
Comment (sigalsax) [inline conjur/wrapper/__init__.py:4]: This module contains wrapper classes for third party libraries

File: test/test_unit_user_logic.py [hunk 24]
```diff
@@ -84,6 +98,7 @@ def test_rotate_other_api_key_returns_new_key(self):
     '''
     Validates that a new personal API key can be returned
     '''
+
```
Comment (sigalsax) [inline test/test_unit_user_logic.py:101]: remove these lines between comments and test

File: test/test_unit_user_logic.py [hunk 25]
```diff
@@ -55,25 +64,30 @@ def test_user_provides_username_can_rotate_anothers_key(self):
...
         mock_user_logic = UserLogic(self.conjurrc_data, self.credentials_from_store, self.client)
-        mock_user_logic.credentials_from_store.load=MagicMock(return_value=CONJURRC_DICT)
+        mock_user_logic.credentials_from_store.load = MagicMock(return_value=CONJURRC_DICT)
         mock_user_logic.extract_credentials_from_credential_store()
         self.assertEquals(mock_user_logic.extract_credentials_from_credential_store(), CONJURRC_DICT)
 
     '''
     Validates that a rotated API key for another user can be returned
     '''
+
```
Comment (sigalsax) [inline test/test_unit_user_logic.py:90]: remove these lines between comments and test

File: test/test_unit_user_logic.py [hunk 26]
```diff
@@ -105,6 +121,7 @@ def test_rotate_personal_api_key_raises_exception_when_unauthorized(self):
     '''
     Raises exception when operation was not able to be completed successfully
     '''
+
```
Comment (sigalsax) [inline test/test_unit_user_logic.py:124]: remove these lines between comments and test

File: conjur/data_object/__init__.py [hunk 27]
```diff
@@ -0,0 +1,14 @@
+"""
+data_object package
```
Comment (sigalsax) [inline conjur/data_object/__init__.py:2]: dont need package

File: conjur/data_object/__init__.py [hunk 28]
```diff
@@ -0,0 +1,14 @@
...
+
+This package contain the data objects of the cli.
```
Comment (sigalsax) [inline conjur/data_object/__init__.py:4]: This module contains the data objects of the client

File: CHANGELOG.md [hunk 29]
```diff
@@ -27,6 +27,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Update version from 0.x.x to 7.0.0 to prep for the GA release
 - Update help screens according to [these guidelines](https://ljfz3b.axshare.com/#id=yokln4&p=conjur_main_help&g=1).
   [cyberark/conjur-api-python3#92](https://github.com/cyberark/conjur-api-python3/issues/92)
+- Directory structure has been refactored. See [design document](https://github.com/cyberark/conjur-api-python3/blob/master/design/general_refactorings.md) for more details.
+  Users that import `Api` and `Client` should change their import statement to  api dir now to the import.
+  This is a breaking change. Users who import the SDK in their projects should change their import statement from 
+  `conjur.client import Client` will change it to `from conjur.api.client import Client`
```
Comment (sigalsax) [inline CHANGELOG.md:33]: will need to change it to..

File: CHANGELOG.md (hunk 12 above)
Comment (sigalsax) [reply CHANGELOG.md:31]: @tzheleznyak please update this whole entry to what i said above

Comment (sigalsax) [reply CHANGELOG.md:31]: with the links

**Review by sigalsax — CHANGES_REQUESTED** [review]
left some comments

File: conjur/data_object/__init__.py (hunk 27 above)
Comment (tzheleznyak) [reply conjur/data_object/__init__.py:2]: Removed

File: conjur/data_object/__init__.py (hunk 28 above)
Comment (tzheleznyak) [reply conjur/data_object/__init__.py:4]: Fixed

File: test/test_unit_user_logic.py (hunk 26 above)
Comment (tzheleznyak) [reply test/test_unit_user_logic.py:124]: Removed

File: test/test_unit_user_logic.py (hunk 25 above)
Comment (tzheleznyak) [reply test/test_unit_user_logic.py:90]: Removed

File: conjur/logic/__init__.py (hunk 21 above)
Comment (tzheleznyak) [reply conjur/logic/__init__.py:7]: This is not a command logic file but it preforms the logic of getting configuration from netrc file. we talked a week ago about this

File: conjur/wrapper/__init__.py (hunk 22 above)
Comment (tzheleznyak) [reply conjur/wrapper/__init__.py:2]: Removed

File: conjur/wrapper/__init__.py (hunk 23 above)
Comment (tzheleznyak) [reply conjur/wrapper/__init__.py:4]: Changed

File: test/test_unit_user_logic.py (hunk 24 above)
Comment (tzheleznyak) [reply test/test_unit_user_logic.py:101]: Removed. Seems my pycharm linter somehow does it. Do you know how to disable it?

File: conjur/api/__init__.py [hunk 30]
```diff
@@ -0,0 +1,8 @@
+"""
+api
+
+Package containing classes that responsible to communicate with conjur server.
```
Comment (sigalsax) [inline conjur/api/__init__.py:4]: that are responsible for communicating with the Conjur server

File: conjur/controller/__init__.py [hunk 31]
```diff
@@ -0,0 +1,14 @@
+"""
+controller
+
+This package contain the controller classes
```
Comment (sigalsax) [inline conjur/controller/__init__.py:4]: contains the Controller classes

File: conjur/data_object/__init__.py [hunk 32]
```diff
@@ -0,0 +1,13 @@
+"""
+data_object
+
+This module contains the data objects of the client
```
Comment (sigalsax) [inline conjur/data_object/__init__.py:4]: module or package?

File: conjur/logic/__init__.py [hunk 33]
```diff
@@ -0,0 +1,13 @@
+"""
+logic
+
+This package contain the logic classes
```
Comment (sigalsax) [inline conjur/logic/__init__.py:4]: contains

File: conjur/util/__init__.py [hunk 34]
```diff
@@ -0,0 +1,5 @@
+"""
+util
+
+This package is for util modules
```
Comment (sigalsax) [inline conjur/util/__init__.py:4]: utils module or package?

File: design/general_refactorings.md [hunk 35]
```diff
@@ -310,8 +311,8 @@ The following is a mapping of builtin and third-party exceptions that are curren
 
     *  The following are errors that can be thrown from `ssl_service.get_certificate` :
 
-      * `TIMEOUT_ERROR` when connection to server fails. We will add an `except` clause and raise `ConnectionToConjurFailedException` with an error message indicating that there was no response from the server.
```
Comment (sigalsax) [inline design/general_refactorings.md:313]: why was this changed? this was correct

File: conjur/util/__init__.py (hunk 34 above)
Comment (tzheleznyak) [reply conjur/util/__init__.py:4]: Each file in the package is a util module. The folder is a package

File: conjur/api/__init__.py (hunk 30 above)
Comment (tzheleznyak) [reply conjur/api/__init__.py:4]: Done

File: conjur/controller/__init__.py (hunk 31 above)
Comment (tzheleznyak) [reply conjur/controller/__init__.py:4]: Done

File: design/general_refactorings.md (hunk 35 above)
Comment (tzheleznyak) [reply design/general_refactorings.md:313]: Fixed

File: conjur/api/api.py (hunk 16 above)
Comment (sigalsax) [reply conjur/api/api.py:16]: @tzheleznyak confirmed the client is still working

File: conjur/util/__init__.py [hunk 36]
```diff
@@ -0,0 +1,5 @@
...
+"""
```
Comment (sigalsax) [inline conjur/util/__init__.py:5]: are there imports import CredentialFromFile?

File: conjur/wrapper/__init__.py [hunk 37]
```diff
@@ -0,0 +1,5 @@
+"""
+wrapper
+
+This module contains wrapper classes for third party libraries
+"""
```
Comment (sigalsax) [inline conjur/wrapper/__init__.py:5]: imports?

File: test/test_unit_user_controller.py [hunk 38]
```diff
@@ -43,13 +46,15 @@ def test_change_password_can_raise_invalid_complexity_error(self):
...
-        mock_user_logic.change_personal_password = MagicMock(side_effect=requests.exceptions.HTTPError(response=mock_response))
+        mock_user_logic.change_personal_password = MagicMock(
+            side_effect=requests.exceptions.HTTPError(response=mock_response))
         with self.assertRaises(InvalidPasswordComplexityException):
             user_controller.change_personal_password()
 
     '''
     Verifies that the user is prompted to input their password and check_password_validity is called once
     '''
+
```
Comment (sigalsax) [inline test/test_unit_user_controller.py:57]: remove this line

File: conjur/util/__init__.py (hunk 34 above)
Comment (sigalsax) [reply conjur/util/__init__.py:4]: did you use the existing Util folder or create a new one? a util folder already exists

Comment (sigalsax) [reply conjur/util/__init__.py:4]: https://github.com/cyberark/conjur-api-python3/tree/refactor-directories/Utils Maybe this should be moved into Conjur

File: conjur/wrapper/__init__.py (hunk 37 above)
Comment (tzheleznyak) [reply conjur/wrapper/__init__.py:5]: Added for ArgumentWrapper.
But HttpWrapper don't have a class so i won't put it here

File: test/test_unit_user_controller.py (hunk 38 above)
Comment (tzheleznyak) [reply test/test_unit_user_controller.py:57]: Removed

File: conjur/util/__init__.py (hunk 36 above)
Comment (tzheleznyak) [reply conjur/util/__init__.py:5]: Added

File: conjur/util/__init__.py (hunk 34 above)
Comment (tzheleznyak) [reply conjur/util/__init__.py:4]: Unified util folders

File: conjur/logic/__init__.py (hunk 33 above)
Comment (tzheleznyak) [reply conjur/logic/__init__.py]: Done

File: conjur/data_object/__init__.py (hunk 32 above)
Comment (tzheleznyak) [reply conjur/data_object/__init__.py:4]: changed to package

**Review by sigalsax — APPROVED** [review]

**Comment by sigalsax** [comment]
@tzheleznyak fantastic work! Please squash and merge so commits will be pretty (:
